tools/clang_tidy/lib/src/options.dart

```diff
@@ -5,6 +5,10 @@
 import 'dart:io' as io show Directory, File, Platform, stderr;
 
 import 'package:args/args.dart';
+import 'package:path/path.dart' as path;
+
+// Path to root of the flutter/engine repository containing this script.
+final String _engineRoot = path.dirname(path.dirname(path.dirname(path.dirname(path.fromUri(io.Platform.script)))));
```

Comment:
If you're computing this here, maybe we don't need the `--repo` command line argument anymore. Or the other way around, if you have the `--repo` argument, this is just the parent directory of that, right?

Reply:
Was there any case where `--repo` needs to handle some other repo directory? I wasn't sure why it was passed in in the first place.

Reply:
This pattern is probably safe here since we're always running the script from source, but it would break if the script were compiled to a snapshot and then the snapshot moved somewhere else before being run.

We can keep this and get rid of the `--repo` argument if that is simpler.

Reply:
> it would break if the script were compiled to a snapshot and then the snapshot moved somewhere else before being run.

If you do that and it doesn't work: Give The Script A Break 🙂

Reply:
Removed `--repo`.